notebooks/text_models/solutions/text_generation_using_transformers.ipynb

```diff
@@ -0,0 +1,911 @@
+{
```

Comment:
<p>Perhaps worth mentioning that this <code>TokenAndPositionEmbedding</code> creates trainable positional embedding instead of the fixed sinusoidal embedding which original Transformer was using.</p><p> </p><p>Also, I'd add what components are contained in <code>TransformerDecoder</code> with a diagram from the paper if appropriate. </p><p>Adding the link to the source code (or Keras document) may be helpful for advanced and curious students.</p><p>https://github.com/keras-team/keras-nlp/blob/v0.6.1/keras_nlp/layers/modeling/transformer_decoder.py#L31</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/GoogleCloudPlatform/asl-ml-immersion/pull/372/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='18'/>

Reply:
<p>Nice suggestion. I have added the hyperlinks and updated the description of the embedding layer. Unfortunately the original paper doesn't have a decoder only model represented and showing both the encoder and the decoder will be confusing so for now, I'll leave out the diagram.</p>